extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authentication/entity/BasicAuthenticatorCredentials.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.authentication.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.druid.security.basic.BasicAuthUtils;
+
+import java.util.Arrays;
+
+public class BasicAuthenticatorCredentials
+{
+  private final byte[] salt;
+  private final byte[] hash;
+  private final int iterations;
+
+  @JsonCreator
+  public BasicAuthenticatorCredentials(
+      @JsonProperty("salt") byte[] salt,
+      @JsonProperty("hash") byte[] hash,
+      @JsonProperty("iterations") int iterations
+  )
+  {
+    this.salt = salt;
+    this.hash = hash;
```

Comment:
Please add ```Preconditions.notNull()``` check for the above two variables.

Reply:
Add preconditions check